src/v/cluster/security_frontend.cc

```diff
@@ -308,4 +309,59 @@ security_frontend::dispatch_delete_acls_to_leader(
       });
 }
 
+/**
+ * For use during cluster creation, if RP_BOOTSTRAP_USER is set
+ * then write a user creation message to the controller log.
+ *
+ * @returns an error code if controller log write failed.  If the
+ *          environment variable is missing or malformed this is
+ *          not considered an error.
+ *
+ */
+ss::future<std::error_code> security_frontend::maybe_create_bootstrap_user() {
+    static const ss::sstring bootstrap_user_env_key{"RP_BOOTSTRAP_USER"};
```

Comment:
@jcsp using static here is probably not ideal since it adds extra code and guard instructions so that safe multi-threaded access can be guaranteed.